core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java

```diff
@@ -98,9 +67,70 @@ public boolean print(byte[] content, JsonNode configuration) {
       log.error(buffer.toString());
       return true;
     }
+  },
+  RAW_TCP("host", "port") {
+
+    @Override
+    public boolean print(byte[] content, JsonNode configuration) {
+      String host = configuration.get("host").asText();
+      int port = configuration.get("port").asInt();
+      try (Socket socket = new Socket(host, port)) {
+        socket.getOutputStream().write(content);
+      } catch (IOException e) {
+        log.error("Failed to print to socket", e);
+        return false;
+      }
+      return true;
+    }
+
+  },
+  ZEBRA_FTP("host", "password") {
+    @Override
+    public boolean print(byte[] content, JsonNode configuration) {
+      return sendFtpFile(configuration.get("host").asText(), "admin", configuration.get("password").asText(),
+          null, content.hashCode() + ".ZPL", content, configuration);
+    }
+
   };
   private static final Logger log = LoggerFactory.getLogger(Backend.class);
 
+  protected static boolean sendFtpFile(String host, String username, String password, String directory, String filename, byte[] content,
+      JsonNode configuration) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "configuration". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)